### ISSUE TITLE
Tempo_bins set to None gives error in Encoding

### ISSUE THREAD

**Issue by ms374-HW** [Natooz/MidiTok#1]
Hi, you have created a really helpful library. I am planning to use it to preprocess some MIDI files for some training (You can check the preliminary implementation [here](https://github.com/ms374-HW/transformer-music)). After your latest release, I ran the code again and I noticed a bug. When running the following piece of code:
```
remi_enc.tokenize_midi_dataset(files_paths, './', midi_valid)
# or the following:
# Converts MIDI to tokens
tokens, track_info = remi_enc.midi_to_tokens(midi)
```
It raises the following error: `TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'`
The full error is below:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-44-a4bbb621f37e> in <module>
     22 
     23 # Converts MIDI files to tokens saved as JSON files
---> 24 remi_enc.tokenize_midi_dataset(files_paths, './', midi_valid)

/opt/conda/lib/python3.7/site-packages/miditok/midi_tokenizer_base.py in tokenize_midi_dataset(self, midi_paths, out_dir, validation_fn, logging)
    264 
    265             # Converting the MIDI to tokens and saving them as json
--> 266             tokens, track_info = self.midi_to_tokens(midi)
    267             with open(PurePath(out_dir, midi_path).with_suffix(".json"), 'w') as outfile:
    268                 json.dump([tokens[0], track_info[0]], outfile)

/opt/conda/lib/python3.7/site-packages/miditok/midi_tokenizer_base.py in midi_to_tokens(self, midi)
     96         tokens = []
     97         for track in midi.instruments:
---> 98             tokens.append(self.track_to_tokens(track))
     99 
    100         track_info = [(int(track.program), track.is_drum) for track in midi.instruments]

/opt/conda/lib/python3.7/site-packages/miditok/midi_tokenizer_base.py in track_to_tokens(self, track)
    110         track.notes.sort(key=lambda x: (x.start, x.pitch))  # sort notes
    111         remove_duplicated_notes(track.notes)  # remove possible duplicated notes
--> 112         events = self.track_to_events(track)  # get distinct events
    113         return self.events_to_tokens(events)
    114 

/opt/conda/lib/python3.7/site-packages/miditok/remi.py in track_to_events(self, track)
     70         current_tempo_idx = 0
     71         current_tempo = self.current_midi_metadata['tempo_changes'][current_tempo_idx].tempo
---> 72         current_tempo = (np.abs(self.tempo_bins - current_tempo)).argmin()
     73         for note in track.notes:
     74             if note.pitch not in self.pitch_range:  # Notes to low or to high are discarded
```
In your latest release, you have added an implementation of tracking tempo changes and current tempo. However, without specifiying tempo bins in additional tokens (using the default settings), `tempo_bins` is set to None, which may be causing this error. I can fork the code and add some conditional statements for keeping track of tempo for default settings if you want. However, I wanted to point this out for future use.

**Comment by Natooz** [comment]
Hi, thanks a lot for pointing this out !
If you want to fix this bug feel free to send a PR.
Replacing None in [this line](https://github.com/Natooz/MidiTok/blob/main/miditok/midi_tokenizer_base.py#L64) by ```np.zeros(1)``` (or something else that works) should solve it with no consequences afterwards, but maybe a more elegant way can probably be found. (I actually decided to initialized ```current_tempo``` in every cases for optimization purpose -> reduce the nb of loop iterations when tracking tempo)

Best,
Nathan

**Comment by Natooz** [comment]
I've fixed this and released 0.1.1, it should be good to go :)

Thanks again,
Nathan

**Comment by ms374-HW** [comment]
I just used the new version and it seems to work alright. Thank you for the prompt reply

Thanks!
Megha
